subprojects/docs/src/docs/release/notes.md

```diff
@@ -90,6 +90,38 @@ This improves the likelihood of up-to-date and build cache hits when any propert
 
 See [the userguide](userguide/more_about_tasks.html#sec:property_file_normalization) for further information.
 
+### Improved cache hits when empty directories are present
+
+For [up-to-date checks](userguide/more_about_tasks.html#sec:up_to_date_checks) and the [build cache](userguide/build_cache.html), Gradle needs to determine if two directory structures contain the same contents.  When a directory contains an empty directory, it is considered to have different contents than an identical directory where the empty directory does not exist.
+
+This may not always be desirable.  There are many cases where only the files in a directory structure may be significant, and an empty directory will have no impact on the outputs of a task.  In such cases, re-executing the task because an empty directory exists is unnecessary as it will only produce the same outputs.
```

Comment:
```suggestion
This may not always be desirable. There are many cases where only the files in a directory structure may be significant, and an empty directory will have no impact on the outputs of a task. In such cases, re-executing the task because an empty directory exists is unnecessary as it will only produce the same outputs.
```

Nit: extra space